src/agent/CMakeLists.txt

```diff
@@ -27,7 +27,7 @@ set(SOURCES
 )
 
 add_library(Agent ${SOURCES})
-target_include_directories(Agent PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(Agent PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include ${CMAKE_CURRENT_SOURCE_DIR}/src)
```

Comment:
If we go down this road, this added directory here should be `PRIVATE`, otherwise it is still visible to everyone that includes this target.